feat(button): add Compact variant examples to Storybook

Varnika060306 opened the pull request:
## 📝 Description

This PR adds a new example to the Chakra UI Storybook demonstrating the `isCompact` prop for the `Button` component.  
The example showcases four compact button variants:
- Compact Default  
- Compact Blue  
- Compact Outline  
- Compact Small  

These additions improve the discoverability and documentation of the `isCompact` prop for developers using Chakra UI.

---

## 🧰 Current behavior (updates)

The current Storybook does not include any visual reference or usage examples of the `isCompact` prop in the Button component.

---

## 🚀 New behavior

This PR introduces a new story titled **"Compact"** under the `Button` section in Storybook.  
It renders multiple compact-styled buttons using the `isCompact` prop, helping users understand its effect.

---

## 🧨 Is this a breaking change (Yes/No):

**No**

---

## ℹ️ Additional Information

- Tests were run with `pnpm test` using `vitest` — all tests passed successfully ✅  
- Visual output was confirmed under `localhost:6006` in Storybook  
- Story file updated: `button.stories.tsx`

isBatak commented:
The recommended way would be to add boolean variant in the button recipe.
You can extend your recipe like this:
```js
export const buttonRecipe = defineRecipe({
  variants: {
     compact: {
       true: {
          paddingInline: "0.5rem",
          paddingBlock: "0.25rem",
          fontSize: "0.875rem",
          height: "auto",
        },
      },
    },
    //...
 });
 ```
 
Use the Chakra UI CLI typegen to generate new types.
 
 Then use it like this:
 
 `<Button compact size="sm" colorScheme="green">Button</Button>`
 
Chakra UI v3 doesn't prefix boolean props with `is` anymore.

I don't think this `compact` prop should be a part of Chakra core, it should stay in the user-land.
What you can contribute is to add docs for the boolean variant.

Varnika060306 commented:
Hey! I’ve implemented the compact: true boolean variant in the button recipe as per your suggestion, and also added a usage example to Storybook.

Changes made:
	•	Added compact boolean variant inside defineRecipe (under variants.true)
	•	Did not use isCompact, as Chakra UI v3 uses raw booleans
	•	Added <Button compact size="sm" /> in the usage story
	•	Kept everything in user-land (not Chakra core)

Still stuck on:
	•	Running Chakra CLI typegen to generate types for the compact prop – I wasn’t sure how to run it correctly inside the monorepo (especially under packages/react).

Would love your help/guidance on how to run that step properly! Let me know if anything else needs fixing. Thanks again
